service/src/main/java/dk/dbc/search/work/presentation/service/FilterResult.java

```diff
@@ -76,19 +79,33 @@ public WorkInformationResponse processWork(WorkInformation work, String agencyId
                                 e.getValue().stream()
                                         .filter(m -> visibleManifestations.contains(m.manifestationId))
                                         .map(ManifestationInformationResponse::from)
-                                        .collect(Collectors.toSet())))
+                                        .collect(toSet())))
                         .filter(e -> !e.getValue().isEmpty())
-                        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-
+                        .collect(toMap(Map.Entry::getKey, Map.Entry::getValue));
         log.debug("dbUnitInformation = {}", dbUnitInformation);
 
+        Set<String> possibleRelations = work.dbRelUnitInformation.values()
+                .stream()
+                .flatMap(Collection::stream)
+                .map(r -> r.manifestationId)
+                .collect(toSet());
+        Set<String> accessibleRelations = solr.getAccessibleRelations(possibleRelations, agencyId, profile, trackingId);
+
+        RelationIndexComputer relationIndexes = new RelationIndexComputer(accessibleRelations, work.dbRelUnitInformation);
+        dbUnitInformation.forEach((unitId, manifestations) -> {
+            int[] indexes = relationIndexes.unitRelationIndexes(unitId);
+            manifestations.forEach(m -> m.relations = indexes);
+        });
+
         WorkInformationResponse wir = WorkInformationResponse.from(work);
         // Flatten the manifestations - with predictable order
         wir.records = new LinkedHashSet<>();
+        wir.relations = relationIndexes.getRelationList();
         dbUnitInformation.values().stream()
                 .flatMap(Collection::stream)
                 .sorted((l, r) -> l.id.compareTo(r.id))
```

Comment:
Ville det give mening at `ManifestationInformationResponse` implementer `Comparable<ManifestationInformationResponse>` med netop denne implementation af `compareTo()`?

Reply:
Jeg ved det ikke rigitgt. Det er mest fordi den kun kigger på manifestationId, som vi i denne kontekst er sikker på er unik, men hvis der var en `compareTo()`, så ville jeg forvente den var lidt mere generisk, end som så. At rækkefølgen på to forskellige manifestationer er deterministisk. (jeg har vægtet for og imod mange gang).